api/nsadm/service.go

```diff
@@ -261,6 +264,46 @@ func (s *service) RemoveNamespaceUser(ctx context.Context, tenantID, username, o
 	return s.store.NamespaceRemoveMember(ctx, tenantID, user.ID)
 }
 
+func (s *service) GetNamespaceByName(ctx context.Context, namespace string) (*models.Namespace, error) {
+	return s.store.NamespaceGetByName(ctx, namespace)
+}
+
+func (s *service) UpdateWebhook(ctx context.Context, url, tenant, owner string) (*models.Namespace, error) {
+	ns, err := s.store.NamespaceGet(ctx, tenant)
+	if err != nil || ns == nil {
+		return nil, ErrNamespaceNotFound
+	}
+
+	user, _, err := s.store.UserGetByID(ctx, owner, false)
+	if err != nil || user == nil {
+		return nil, ErrUserNotFound
+	}
+
+	if ns.Owner != user.ID {
+		return nil, ErrUnauthorized
+	}
```

Comment:
This block seems repeated in many places. I believe it'd reduce a lot of code here and in other methods.

Could you add a commit before this commit reworking the existing code to avoid this code duplication, and then rework this commit to use it?